src/services/TaskService.php

```diff
@@ -42,4 +48,48 @@ public function createTask(TaskCreateForm $form)
         }
         return true;
     }
+    public function createReply(ReplyForm $form, $taskId)
+    {
+        if (!$form->validate()) {
+            return null;
+        }
+        $reply = new Replies();
+        $reply->task_id = $taskId;
+        $reply->user_id = Yii::$app->user->id;
+        $reply->description = $form->comment;
+        $reply->price = $form->price;
+        return $reply->save();
+    }
+
+    public function completeTask(CompletionForm $form, $taskId)
+    {
+        if (!$form->validate()) {
+            return null;
+        }
+        $task = Tasks::findOne($taskId);
+        if (!$task) {
+            throw new NotFoundHttpException("Задания с id $taskId не существует");
+        }
+        $review = new Reviews();
+        if ($form->isComplete === "yes") {
+            $task->status = Task::STATUS_COMPLETED;
+            $task->save();
+            $review->task_id = $taskId;
+            $review->user_id = $task->contractor_id;
+            $review->review = $form->review;
+            $review->rating = $form->rating;
+            $review->task_completion_status = 'yes';
+            return $review->save();
+        } else {
+            $task->status = Task::STATUS_FAILED;
+            $task->save();
+            $review->task_id = $taskId;
+            $review->user_id = $task->contractor_id;
+            $review->review = $form->review;
+            $review->rating = $form->rating;
+            $review->task_completion_status = 'no';
+            return $review->save();
+        }
```

Comment:
Вот что я имел в виду, говоря о том, что нужно убрать дублирование кода:

```suggestion
        $task->status = $form->isComplete === 'yes' ? Task::STATUS_COMPLETED : Task::STATUS_FAILED;
        $task->save();

        $review->task_id = $taskId;
        $review->user_id = $task->contractor_id;
        $review->review = $form->review;
        $review->rating = $form->rating;
        $review->task_completion_status = $form->isComplete === 'yes' ? 'yes' : 'no';

        return $review->save();
```